Library/Homebrew/mktemp.rb

```diff
@@ -40,7 +46,15 @@ def to_s
   end
 
   def run
-    @tmpdir = Pathname.new(Dir.mktmpdir("#{@prefix.tr "@", "AT"}-", HOMEBREW_TEMP))
+    prefix_name = @prefix.tr "@", "AT"
+    if @retain_in_cache
+      source_dir = "#{HOMEBREW_CACHE}/Sources/#{prefix_name}"
```

Comment:
```suggestion
      source_dir = HOMEBREW_CACHE/"Sources"/prefix_name
```
A bit nicer to me, but optional, since I think interpolation is the pattern in nearby code. (Probably stick to interpolation if that's what you see nearby.)